### PR TITLE
Add hidden property to error

### PR DESCRIPTION
**Pull request by XiaofuHuang**
[AB#249217](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/249217)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6215)

### CODE REVIEW

File: src/docfx/config/ConfigLoader.cs [hunk 1]
```diff
@@ -215,5 +225,29 @@ where key.StartsWith("DOCFX_")
...
+
+            if (validationRulesObject is SourceInfo<string> validationRulesSourceInfo)
+            {
+                var physicalMarkdownValidationRulesPath = ContentValidator.GetValidationPhysicalFilePath(fileResolver, validationRulesSourceInfo);
+                var contentRules = JsonConvert.DeserializeObject<Dictionary<string, ValidationRules>>(File.ReadAllText(physicalMarkdownValidationRulesPath));
+
+                var customRules = contentRules
+                    .SelectMany(attributeRules => attributeRules.Value.Rules)
+                    .Where(contentRule => contentRule.PullRequestOnly)
+                    .ToDictionary(contentRule => contentRule.Code, _ => new { PullRequestOnly = true });
```
Comment (OsmondJiang) [inline src/docfx/config/ConfigLoader.cs:246]: in the future, we can control docfx error codes at here

File: src/docfx/config/ops/OpsMetadataRuleConverter.cs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 using System;
 using System.Collections.Generic;
+using System.Data;
```
Comment (OsmondJiang) [inline src/docfx/config/ops/OpsMetadataRuleConverter.cs:6]: remove

File: src/docfx/config/ConfigLoader.cs [hunk 3]
```diff
@@ -215,5 +225,29 @@ where key.StartsWith("DOCFX_")
...
+            // Fill in CustomRules from content validation rule
+            var result = new JObject();
+            var errors = new List<Error>();
+
+            var (validationRulesError, validationRulesObject) = JsonUtility.ToObject(markdownValidationRules, typeof(SourceInfo<string>));
+            errors.AddRange(validationRulesError);
+
+            if (validationRulesObject is SourceInfo<string> validationRulesSourceInfo)
+            {
+                var physicalMarkdownValidationRulesPath = ContentValidator.GetValidationPhysicalFilePath(fileResolver, validationRulesSourceInfo);
```
Comment (OsmondJiang) [inline src/docfx/config/ConfigLoader.cs:240]: use FileResolver to read content

File: src/docfx/lib/log/Error.cs [hunk 4]
```diff
@@ -91,10 +96,11 @@ public string ToString(ErrorLevel level, SourceMap? sourceMap)
             var file = originalPath == null ? FilePath?.Path : originalPath;
             var date_time = DateTime.UtcNow;
             var log_item_type = "user";
+            var hidden = Hidden;
 
             return originalPath == null
-                ? JsonUtility.Serialize(new { message_severity, log_item_type, Code, Message, file, line, end_line, column, end_column, date_time })
-                : JsonUtility.Serialize(new { message_severity, log_item_type, Code, Message, file });
+                ? JsonUtility.Serialize(new { message_severity, log_item_type, Code, Message, file, line, end_line, column, end_column, hidden, date_time })
```
Comment (OsmondJiang) [inline src/docfx/lib/log/Error.cs:102]: pronly?

File: test/docfx.Test/lib/JsonSchemaTest.cs [hunk 5]
```diff
@@ -290,85 +290,87 @@ public void TestJsonSchemaConformance(string description, string schemaText, str
...
-            "{'message_severity':'suggestion','log_item_type':'user','code':'key1-attribute-deprecated','message':'Deprecated attribute: 'key1', use 'key2' instead.','file':'file','line':1,'end_line':1,'column':10,'end_column':10}")]
+            "{'message_severity':'suggestion','log_item_type':'user','code':'key1-attribute-deprecated','message':'Deprecated attribute: 'key1', use 'key2' instead.','file':'file','line':1,'end_line':1,'column':10,'end_column':10,'hidden':false}")]
         [InlineData("{'properties': {'keys': {'precludes': [['key1', 'key2']]}}, 'customRules': {'key1': {'precluded-attributes': {'severity': 'error'}}}}", "{'keys' : {'key1': 1, 'key2': 2}}",
-            "{'message_severity':'error','log_item_type':'user','code':'precluded-attributes','message':'Only one of the following attributes can exist: 'key1', 'key2'.','file':'file','line':1,'end_line':1,'column':11,'end_column':11}")]
+            "{'message_severity':'error','log_item_type':'user','code':'precluded-attributes','message':'Only one of the following attributes can exist: 'key1', 'key2'.','file':'file','line':1,'end_line':1,'column':11,'end_column':11,'hidden':false}")]
```
Comment (OsmondJiang) [inline test/docfx.Test/lib/JsonSchemaTest.cs:360]: we can remove hidden=false using regex

Comment (XiaofuHuang) [reply test/docfx.Test/lib/JsonSchemaTest.cs]: The pr only is hidden when its value is false.

File: src/docfx/lib/log/Error.cs [hunk 6]
```diff
@@ -28,8 +28,10 @@ internal class Error
 
         public int EndColumn { get; }
 
-        public Error(ErrorLevel level, string code, string message, SourceInfo? source, string? name = null)
-            : this(level, code, message, source?.File, source?.Line ?? 0, source?.Column ?? 0, source?.EndLine ?? 0, source?.EndColumn ?? 0, name)
+        public bool PrOnly { get; }
```
Comment (yufeih) [inline src/docfx/lib/log/Error.cs:31]: PullRequestOnly?

File: test/docfx.Test/data/validation/schema.json [hunk 7]
```diff
@@ -5189,7 +5189,7 @@
       "missing-attribute": {
         "severity": "suggestion",
         "code": "ms-topic-missing",
-        "additionalMessage": "Add a valid value to indicate the article type."
+        "additionalMessage": "Add a valid value to indicate the article type.",
```
Comment (mingwli) [inline test/docfx.Test/data/validation/schema.json:5192]: ,

**Pull request merged**